sympy/core/add.py

```diff
@@ -216,10 +216,28 @@ def flatten(cls, seq):
                         # e.g. 3 + ...
         order_factors = []
 
+        saw_inf = None  # flag to consult when encountering zoo
+
         extra = []
 
+        _nan = False
+
+        tensAdd = None
+
+        oocoeff = S.Infinity, S.ComplexInfinity, S.NegativeInfinity, S.ImaginaryUnit*S.Infinity, S.NegativeOne*S.ImaginaryUnit*S.Infinity
```

Comment:
We should check the performance implications of this change. For instance, does constructing these objects and checking against them in every Add.flatten call slow it down?